src/test/test-sleep.c

```diff
@@ -18,13 +18,41 @@
   along with systemd; If not, see <http://www.gnu.org/licenses/>.
 ***/
 
+#include <linux/fiemap.h>
 #include <stdio.h>
 
+#include "fd-util.h"
 #include "log.h"
 #include "sleep-config.h"
 #include "strv.h"
 #include "util.h"
 
+static int test_fiemap(void) {
+        _cleanup_free_ struct fiemap *fiemap = NULL;
+        _cleanup_close_ int fd = -1;
+        int r;
+        const char *device = ABS_BUILD_DIR "/systemd-sleep";
```

Comment:
This would work most of the time, but we also allow tests to be installed. So the test binary should work even if the build directory is removed. Hence getting access to the files in the build tree is rather complicated, see get_testdata_dir() if interested. Fortunately, here we can just use argv[0], it'll work just as well.

I'll push a follow-up patch that allows testing on arbitrary files and uses argv[0]. Please squash it in in the next round.